compositor_render/src/transformations/web_renderer/shader.rs

```diff
@@ -0,0 +1,199 @@
+                    }],
+                });
+
+        let pipeline_layout =
+            wgpu_ctx
+                .device
+                .create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+                    label: Some("Web renderer pipeline layout"),
+                    bind_group_layouts: &[
+                        &textures_bgl,
+                        &wgpu_ctx.shader_parameters_bind_group_layout,
+                        &sampler.bind_group_layout,
+                    ],
+                    push_constant_ranges: &[wgpu::PushConstantRange {
+                        stages: wgpu::ShaderStages::VERTEX_FRAGMENT,
+                        range: 0..CommonShaderParameters::push_constant_size(),
+                    }],
+                });
+
+        let pipeline = wgpu_ctx
+            .device
+            .create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+                label: Some("shader transformation pipeline :^)"),
+                depth_stencil: None,
+                primitive: wgpu::PrimitiveState {
+                    conservative: false,
+                    cull_mode: Some(wgpu::Face::Back),
+                    front_face: wgpu::FrontFace::Ccw,
+                    polygon_mode: wgpu::PolygonMode::Fill,
+                    topology: wgpu::PrimitiveTopology::TriangleList,
+                    strip_index_format: None,
+                    unclipped_depth: false,
+                },
+                vertex: wgpu::VertexState {
+                    buffers: &[Vertex::LAYOUT],
+                    module: &shader_module,
+                    entry_point: VERTEX_ENTRYPOINT_NAME,
+                },
+                fragment: Some(wgpu::FragmentState {
+                    module: &shader_module,
+                    entry_point: FRAGMENT_ENTRYPOINT_NAME,
+                    targets: &[Some(wgpu::ColorTargetState {
+                        format: wgpu::TextureFormat::Rgba8Unorm,
+                        write_mask: wgpu::ColorWrites::all(),
+                        blend: Some(wgpu::BlendState::ALPHA_BLENDING),
+                    })],
+                }),
+                layout: Some(&pipeline_layout),
+                multisample: wgpu::MultisampleState {
+                    count: 1,
+                    mask: !0,
+                    alpha_to_coverage_enabled: false,
+                },
+                multiview: None,
+            });
+
+        scope.pop(&wgpu_ctx.device)?;
+
+        Ok(Self {
+            wgpu_ctx: wgpu_ctx.clone(),
+            pipeline,
+            textures_bgl,
+            sampler,
+        })
+    }
+
+    pub(super) fn render(
+        &self,
+        params: &wgpu::BindGroup,
+        textures: &[Option<&Texture>],
+        target: &NodeTextureState,
+        pts: Duration,
+        clear_color: Option<wgpu::Color>,
+    ) {
+        let mut texture_views: Vec<&wgpu::TextureView> = textures
```

Comment:
make the same change as in the layout node, and pass only one texture per draw call. If you want to do that in follow-up PR it's fine, but I think there is not much point in separating that.